src/parse.c

```diff
@@ -725,12 +739,59 @@ static ident_t loc_to_ident(const loc_t *loc)
    return ident_new(tb_get(tb));
 }
 
+static ident_t get_implicit_label(tree_t t, const loc_t *loc)
+{
+   int *cnt;
+   char c;
+
+   switch (tree_kind(t)) {
+   case T_PROCESS:
+   case T_CONCURRENT:
+      cnt = &(imp_label_cnts.proc);
+      c = 'P';
+      break;
+
+   case T_FOR:
+   case T_WHILE:
+      cnt = &(imp_label_cnts.loop);
+      c = 'L';
+      break;
+
+   case T_WAIT:
+   case T_ASSERT:
+   case T_IF:
+   case T_NULL:
+   case T_RETURN:
+   case T_CASE:
+   case T_EXIT:
+   case T_NEXT:
+   case T_SIGNAL_ASSIGN:
+   case T_VAR_ASSIGN:
+   case T_PROT_PCALL:
+   case T_PCALL:
+   case T_COND_ASSIGN:
+   case T_SELECT:
+      cnt = &(imp_label_cnts.stmt);
+      c = 'S';
+      break;
+
+   default:
+      return loc_to_ident(loc);
+   }
+
+   char name[22];
+   memset(name, 0, sizeof(char) * 22);
```

Comment:
It's not necessary to zero the buffer here.

Reply:
removed.